src/core/src/model.cpp

```diff
@@ -199,6 +199,7 @@ void ov::Model::prerequirements(bool detect_variables, bool detect_parameters) {
         OPENVINO_ASSERT(param != nullptr, "Model is incorrect! Some Parameter operation equals to nullptr.");
     }
 
+    OPENVINO_ASSERT(m_results.size() != 0, "Model is incorrect! No Result operations found.");
     for (const auto& result : m_results) {
```

Comment:
@ilya-lavrenov no, the problem is that `ov::Model` may be created w/o parameters or results, it can be compiled by, at least, CPU, but when trying to call property like "get_network_name", it lead so segfault 


For the moment we think that need to check Results in `CompiledModel`, because `ov::Model` have an API to add/remove Results, so it may not have Results at object creation